InvokePesterStub.ps1

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env pwsh
+
+<#
+.SYNOPSIS
+    Stub around Invoke-Pester command used by VSCode PowerShell extension.
+.DESCRIPTION
+    The stub checks the version of Pester and if >= 4.6.0, invokes Pester
+    using the LineNumber parameter (if specified). Otherwise, it invokes
+    using the TestName parameter (if specified). If the All parameter
+    is specified, then all the tests are invoked in the specifed file.
+    Finally, if none of these three parameters are specified, all tests
+    are invoked and a warning is issued indicating what the user can do
+    to allow invocation of individual Describe blocks.
+.EXAMPLE
+    PS C:\> .\InvokePesterStub.ps1 ~\project\test\foo.tests.ps1 -LineNumber 14
+    Invokes a specific test by line number in the specified file.
+.EXAMPLE
+    PS C:\> .\InvokePesterStub.ps1 ~\project\test\foo.tests.ps1 -TestName 'Foo Tests'
+    Invokes a specific test by test name in the specified file.
+.EXAMPLE
+    PS C:\> .\InvokePesterStub.ps1 ~\project\test\foo.tests.ps1 -All
+    Invokes all tests in the specified file.
+.INPUTS
+    None
+.OUTPUTS
+    None
+#>
+param(
+    # Specifies the path to the test script.
+    [Parameter(Position=0, Mandatory)]
+    [ValidateNotNullOrEmpty()]
+    [string]
+    $ScriptPath,
+
+    # Specifies the name of the test taken from the Describe block's name.
+    [Parameter()]
+    [string]
+    $TestName,
+
+    # Specifies the starting line number of the DescribeBlock.  This feature requires
+    # Pester 4.6.0 or higher.
+    [Parameter()]
+    [ValidatePattern('\d*')]
+    [string]
+    $LineNumber,
+
+    # If specified, executes all the tests in the specified test script.
+    [Parameter()]
+    [switch]
+    $All
+)
+
+$pesterModule = Microsoft.PowerShell.Core\Get-Module Pester
+if (!$pesterModule) {
+    Write-Output "Importing Pester module..."
+    $pesterModule = Microsoft.PowerShell.Core\Import-Module Pester -ErrorAction Ignore -PassThru
+    if (!$pesterModule) {
+        # If we still don't have an imported Pester module, that is (most likely) because Pester is not installed.
+        Write-Warning "Failed to import the Pester module. You must install Pester to run or debug Pester tests."
+        Write-Warning "You can install Pester by executing: Install-Module Pester -Scope CurrentUser -Force"
+        return
+    }
+}
+
+if ($All) {
+    Pester\Invoke-Pester -Script $ScriptPath -PesterOption @{IncludeVSCodeMarker=$true}
+}
+elseif (($LineNumber -match '\d+') -and ($pesterModule.Version -ge '4.6.0')) {
```

Comment:
Not sure if `return`s instead of `else`s are worth contemplating, but could reduce complexity if the script grows

Reply:
I we should start calling you 'Rob "return instead of else" Holt' 😉

Alternatively a `switch ($true)` and `break`s can be used here I think.

Reply:
One of my early mentor's called this 'else'-less programming

Reply:
While `switch ($true)` could be made to work, it seems a bit "unusual.  The script does use one early return - if Pester has not been installed.  But for the rest, it is pretty compact and understandable as-is I think.

Reply:
> `switch ($true)`

*shudder*

Reply:
> One of my early mentors called this 'else'-less programming

I've had my brain poisoned by `if`-less programming, in languages where `if` is considered clumsy. The big takeaway for me was that being able to read down the page/screen following a process of elimination meant I was much less likely to introduce a bug by trying to juggle an implicit union state from multiple possible inputs.